### DISCUSSION TITLE
Failed adding main jpg file with no specific error

### DISCUSSION THREAD

**Discussion by rkuzmenkov** [photoprism/photoprism#3156] [General]
Hi,

I have one photo which is constantly failing on indexing with the following log records:
> 2023-01-31 18:37:43 ERRO index: failed adding main jpg file 'Misc Photo/2013/2013-08-25_15-54_06.JPG'
> 2023-01-31 18:37:43 ERRO index: %!s(<nil>) in 'Misc Photo/2013/2013-08-25_15-54_06.JPG' (find or create photo)
> 2023-01-31 18:37:43 INFO index: matched 1 label with 2013-08-25_15-54_06.JPG [727.48731ms]

Photoptism build: 221118-e58fee0fb

What I've tried:
* renaming the file;
* resizing the photo in Windows;
* rotating image in Windows;
* doing convert with ImageMagick.

Nothing changes.

Also, I tried to import it to demo.photoprism.app. And it worked.

Thank you in advance for any suggestions.

**Comment by lastzero** [comment]
Debug logs might help. Also check permissions and that you have enough memory / swap.

**Comment by rkuzmenkov** [comment]
Debug log gives not much info as well:
{"log":"time=\"2023-01-31T15:22:27Z\" level=debug msg=\"media: created no new thumbnails for 'Misc Photo/2013/2013-08-25_15-54-06.JPG' [3.872493ms]\"\n","stream":"stderr","time":"2023-01-31T15:22:27.836996487Z"}
{"log":"time=\"2023-01-31T15:22:28Z\" level=info msg=\"media: 2013-08-25_15-54-06.JPG was taken at 2013-08-25 11:54:05 +0000 UTC (meta)\"\n","stream":"stderr","time":"2023-01-31T15:22:28.016601076Z"}
{"log":"time=\"2023-01-31T15:22:32Z\" level=debug msg=\"photo: changing location of 'Misc Photo/2013/2013-08-25_15-54-06' from zz to s2:40c1af955efc\"\n","stream":"stderr","time":"2023-01-31T15:22:32.005720968Z"}
{"log":"time=\"2023-01-31T15:22:32Z\" level=error msg=\"index: %!s(\u003cnil\u003e) in 'Misc Photo/2013/2013-08-25_15-54-06.JPG' (find or create photo)\"\n","stream":"stderr","time":"2023-01-31T15:22:32.01078561Z"}
{"log":"time=\"2023-01-31T15:22:32Z\" level=error msg=\"index: failed adding main jpg file 'Misc Photo/2013/2013-08-25_15-54-06.JPG'\"\n","stream":"stderr","time":"2023-01-31T15:22:32.010819565Z"}

Permissions are ok and the same as for other files.
Memory usage 32.45% (1.30 GiB of 4.00 GiB)
SWAP usage 0.56% (22.89 MiB of 4.00 GiB)
~18k other files were imported with no issues.

Also, I've just updated it to the latest preview. And the error is still there.

**Reply by lastzero** [reply]
What storage device & filesystem?

**Reply by rkuzmenkov** [reply]
Storage is Synology NAS mounted as a NFS share to the originals directory.

The file has the same Linux permissions as all other files. And it's readable/writable.
Interestingly, this only file had owner 'admin' in the Synology share. Apparently, the reason is that it was created by 'convert' in the host. And I have all users mapped to admin for the NFS share.
But I've changed the owner to the same one as for other files, which didn't help.

Also, I've tried to do Import - Upload for this and several other files. Other files are successfully imported. This one has the same error.

**Reply by lastzero** [reply]
If you can't reproduce the problem without NFS and with files owned by the user PhotoPrism is running as, I would guess it is a file system, permission or network issue. Our free community support does not provide troubleshooting help for these types of issues, but feel free to post the solution if you were able to figure it out. Maybe other community members can help as well. Ideally we can also show a more helpful error message once it's clear where this happens.

**Reply by lastzero** [reply]
One thing to check are the permissions of related sidecar and cache files. The main media file may not necessarily be the problem.

**Reply by rkuzmenkov** [reply]
First of all, this issue is not related to permissions. I removed mounted NFS shares. Completely purged database, sidecar, thumbnails, etc.
Again. All files are successfully processed both from Originals and Import. This only file failed.

I've tried doing fresh installs using Debian/Ubuntu, installing from sources and using docker-compose, MariaDB and Sqlite.
The file can be successfully imported in any configuration not using MariaDB.

So the issue is in adding file data to DB.

I didn't find a way for now to trace DB SQL queries. Is there any way to log DB SQL errors somehow?

**Reply by rkuzmenkov** [reply]
It looks like I found the root cause.
The photo metadata has altitude = 4294967284. It gives an error: Out of range value for column 'photo_altitude'.

It would be great if not to check the validity of the metadata fields then show SQL errors in debug mode.

Upd: digiKam shows altitude = -12 for the photo.

**Reply by lastzero** [reply]
Invalid or extremely large metadata that does not fit in the database table can cause this. However, a negative altitude should not really be a problem, because that is to be expected (unless it's a glitch in the schema).

**Reply by lastzero** [reply]
In order to better handle this, it's best to send us a file sample for testing: https://photoprism.app/contact#file-samples

**Reply by graciousgrey** [reply]
This should be fixed with https://github.com/photoprism/photoprism/issues/3182 :)
